packages/tools/devtools/devtools-core/src/DecomposedContainer.ts

```diff
@@ -0,0 +1,115 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TypedEventEmitter } from "@fluid-internal/client-utils";
+import type { IAudience } from "@fluidframework/container-definitions";
+import type {
+	AttachState,
+	IContainerEvents,
+	ICriticalContainerError,
+} from "@fluidframework/container-definitions/internal";
+import { ConnectionState } from "@fluidframework/container-loader";
+import type { IEventProvider } from "@fluidframework/core-interfaces";
+import type { IFluidDataStoreRuntime } from "@fluidframework/datastore-definitions/internal";
+
+/**
+ * A lightweight abstraction of {@link @fluidframework/container-definitions#IContainer} that provides
+ * only the essential properties and methods needed for Fluid DevTools functionality.
+ */
+export interface DecomposedIContainer extends IEventProvider<IContainerEvents> {
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.audience}
+	 */
+	readonly audience: IAudience;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.clientId}
+	 */
+	readonly clientId?: string | undefined;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.attachState}
+	 */
+	readonly attachState: AttachState;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.connectionState}
+	 */
+	readonly connectionState: ConnectionState;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.closed}
+	 */
+	readonly closed: boolean;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.connect}
+	 */
+	connect?(): void;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.disconnect}
+	 */
+	disconnect?(): void;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.close}
+	 */
+	close?(error?: ICriticalContainerError): void;
+}
+
+/**
+ * Implementation of {@link DecomposedIContainer} that wraps an {@link IFluidDataStoreRuntime}.
+ * This class provides a bridge between {@link IFluidDataStoreRuntime} and the devtools system by exposing runtime properties and events as {@link IContainer} interfaces.
+ *
+ * @alpha
+ */
+export class DecomposedContainer
```

Comment:
Rather than exporting this class, it might be a bit nicer to make this class private to this module and add a factory function for mapping the datastore runtime to our decomposed container interface. Then no one has to use the terrible naming suggestion I made in [this comment](https://github.com/microsoft/FluidFramework/pull/24885/files#r2196089114).